projects/graph/src/graph.py

```diff
@@ -5,5 +5,35 @@
 
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
+
     def __init__(self):
-        pass  # TODO
+        self.vertices = {}
+
+    def add_vertex(self, vertex):
+        if vertex not in self.vertices:
+            self.vertices[vertex] = set()
+            return True
+        else:
+            return False
+
+    def add_edge(self, v1, v2):
+        if v1 in self.vertices and v2 in self.vertices:
+            self.vertices[v1].add(v2)
+            self.vertices[v2].add(v1)
+            return True
+        else:
+            if v1 not in self.vertices:
+                raise Exception(f"Vertex {v1} is nonexistant!")
```

Comment:
This is good! If you want to be more specific you could also do ```raise IndexError``` since we're discussing a non-existant entry into a dict.

Reply:
Also love that you're still using the string formatting that we learned about a few weeks ago.